### ISSUE TITLE
Arduino.h: No such file or directory

### ISSUE THREAD

**Issue by tsctrl** [espressif/arduino-esp32#4609]
I have idf 4.2 relese branch with arduino lib 4.2 release branch as component manually added under component/arduino folder inside my project.
* i have issue with variable undefined where i have to remove all this lib from my project to build:
libraries/LITTLEFS/
libraries/WiFiProv/

This is mostly because my library are pointing to idf installation folder. But the variables are in arduino component sdk folder.
Not sure how to fix, should i override idf components or how to make my arduino library referring to the sdk under the arduino component folder?
Once i am able to build, i created blink project where the project works and able to fire the tiny led at the esp32 devkitc board.

I created new libraries with .cpp (in src) and .h (in includes folder). 
.h files are c++ classes with pubic functions.
.cpp is only function implementation that includes .h classes.
.h will be included in the app_main.cpp main class.

* basically standard c++ arduino library implementation.

system( Hardware, computer, O/S, core version, environment):
Windows. ESP32 devkit C.

Describe what is failing:
I am not able to include any Arduino library into my cpp/h files in my components. I think i miss some declaration somewhare?

Build error:
../components/test/include/test.h:19:10: fatal error: Arduino.h: No such file or directory
fatal error: SPI.h: No such file or directory etc

test.h:
#ifndef COMPONENTS_TEST_INCLUDE_TEST_H_
#define COMPONENTS_TEST_INCLUDE_TEST_H_
#include <Arduino.h>
#include "DS3231.h"
#include "EEPROM.h"
class Abc {
....
}
#endif

If i remove #include <Arduino.h, error will be on #include "DS3231.h". So on

CMakeLists:
idf_component_register(
    SRCS "src/test.cpp" 
    INCLUDE_DIRS "./include"
)
